pages/transactions/[id].js

```diff
@@ -18,20 +19,30 @@ export default function TransactionDetailsPage({
   handleDeleteTransaction,
   successMessage,
 }) {
+  const { data: categories = [] } = useSWR(`/api/categories`, fetcher);
```

Comment:
you already specified a default fetcher with the SWRConfig, you don't need this fetcher function.